src/api/firebase.js

```diff
@@ -140,6 +144,28 @@ export async function createList(userId, userEmail, listName) {
 	}
 }
 
+/**
+ * Deletes the currently selected list, returns boolean for user feedback.
+ * @param {string} userEmail The email of the user that is currently logged in.
+ * @param {string} listPath The path to the list to share.
+ * @param {string} userId The id assigned to the user within firebase (uid)
+ */
+export async function deleteList(userEmail, listPath) {
+	const listDocumentRef = doc(db, listPath);
+	const userDocumentRef = doc(db, 'users', userEmail);
+	try {
+		// removing list from current user's lists:
+		deleteDoc(listDocumentRef);
+		updateDoc(userDocumentRef, {
+			sharedLists: arrayRemove(listDocumentRef),
+		});
+		return true;
```

Comment:
Thank you! Talking through this issue at our office hours on Thurs helped me understand that we weren't destroying other lists when we removed them from the SharedLists for each user, so that check I had before was sort of overthinking things.

Reply:
praise: I'm glad you brought this up during office hours and we were able to all talk through the potential consequences of the actions being considered. Good job taking the user experience into account.